extensions/ql-vscode/src/data-extensions-editor/queries/csharp.ts

```diff
@@ -2,33 +2,52 @@ import { Query } from "./query";
 
 export const fetchExternalApisQuery: Query = {
   mainQuery: `/**
- * @name Usage of APIs coming from external libraries
- * @description A list of 3rd party APIs used in the codebase.
- * @tags telemetry
- * @id cs/telemetry/fetch-external-apis
- */
+* @name Usage of APIs coming from external libraries
+* @description A list of 3rd party APIs used in the codebase.
+* @tags telemetry
+* @id cs/telemetry/fetch-external-apis
+*/
+
+import csharp
+import ExternalApi
+
+private Call aUsage(ExternalApi api) {
+ result.getTarget().getUnboundDeclaration() = api
+}
+
+private boolean isSupported(ExternalApi api) {
+ api.isSupported() and result = true
+ or
+ not api.isSupported() and
+ result = false
+}
+
+from ExternalApi api, string apiName, boolean supported, Call usage
+where
+ apiName = api.getApiName() and
+ supported = isSupported(api) and
+ usage = aUsage(api)
+select apiName, supported, usage
+`,
+  usagesQuery: `/**
+* @name Usage of APIs coming from external libraries
+* @description A list of 3rd party APIs used in the codebase.
+* @kind problem
+* @id cs/telemetry/fetch-external-api-usages
+*/
+
+import csharp
+import ExternalApi
+
+private Call aUsage(ExternalApi api) {
+ result.getTarget().getUnboundDeclaration() = api
+}
 
- import csharp
- import semmle.code.csharp.dataflow.internal.FlowSummaryImpl as FlowSummaryImpl
- import ExternalApi
- 
- private Call aUsage(ExternalApi api) {
-   result.getTarget().getUnboundDeclaration() = api
- }
- 
- private boolean isSupported(ExternalApi api) {
-   api.isSupported() and result = true
-   or
-   not api.isSupported() and
-   result = false
- }
- 
- from ExternalApi api, string apiName, boolean supported, Call usage
- where
-   apiName = api.getApiName() and
-   supported = isSupported(api) and
-   usage = aUsage(api)
- select apiName, supported, usage
+from ExternalApi api, string apiName, Call usage
+where
+ apiName = api.getApiName() and
+ usage = aUsage(api)
+select usage, apiName
```

Comment:
I am ok to do it like this for now. But it _feels_ like we should be able to reuse the previous query as it selects all the same data.

Reply:
I found a way to use a single query for this, so I'll change it to use a single query for both retrieving the external APIs and for retrieving the usages.